tasty/src/dotty/tools/tasty/TastyFormat.scala

```diff
@@ -62,17 +62,20 @@ Standard-Section: "ASTs" TopLevelStat*
                   IMPORT         Length qual_Term Selector*                        -- import qual selectors
                   EXPORT         Length qual_Term Selector*                        -- export qual selectors
   ValOrDefDef   = VALDEF         Length NameRef type_Term rhs_Term? Modifier*      -- modifiers val name : type (= rhs)?
-                  DEFDEF         Length NameRef TypeParam* Params* returnType_Term
-                                        rhs_Term? Modifier*                        -- modifiers def name [typeparams] paramss : returnType (= rhs)?
+                  DEFDEF         Length NameRef Param* returnType_Term rhs_Term?
+                                        Modifier*                                  -- modifiers def name [typeparams] paramss : returnType (= rhs)?
   Selector      = IMPORTED              name_NameRef                               -- name, "_" for normal wildcards, "" for given wildcards
                   RENAMED               to_NameRef                                 -- => name
                   BOUNDED               type_Term                                  -- type bound
 
   TypeParam     = TYPEPARAM      Length NameRef type_Term Modifier*                -- modifiers name bounds
-  Param         = PARAM          Length NameRef type_Term rhs_Term? Modifier*      -- modifiers name : type (= rhs_Term)?. `rhsTerm` is present in the case of an aliased class parameter
-                  PARAMEND                                                         -- ends a parameter clause
-                                                																   -- needed if previous parameter clause is empty or another parameter clause follows
-  Template      = TEMPLATE       Length TypeParam* Param* parent_Term* Self? Stat* -- [typeparams] paramss extends parents { self => stats }, where Stat* always starts with the primary constructor.
+  TermParam     = PARAM          Length NameRef type_Term rhs_Term? Modifier*      -- modifiers name : type (= rhs_Term)?. `rhsTerm` is present in the case of an aliased class parameter
+                  EMPTYCLAUSE                                                      -- an empty parameter clause ()
+                  SPLITCLAUSE                                                      -- splits two non-empty parameter clauses of the same kind
+  Param         = TypeParam
+                  TermParam
+  Template      = TEMPLATE       Length TypeParam* TermParam* parent_Term* Self?
```

Comment:
Should we future proof this definition by allowing `class Foo[T](a: A)[U](a: B)` internally in TASTY?
```
  Template      = TEMPLATE       Length Param* parent_Term* Self?
```

Reply:
I think that's not on the agenda right now. I see all sorts of complications arising from this. 

So I think all of this can wait until we go to the next major redesign for Tasty in 3.T